### PR TITLE
data model: add vernacular language unimarc support

### PR DESCRIPTION
**Pull request by reropag**
Co-Authored-by: Gianni Pante <gianni.pante@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: rero_ils/dojson/utils.py [hunk 1]
```diff
@@ -212,9 +242,58 @@ def get_subfields(self, field, code=None):
...
+            'value': clean_punctuation(label, punct, spaced_punct).strip()
+        }]
+        try:
+            alt_gr = self.alternate_graphic[tag][link]
+            subfield = self.get_subfields(alt_gr['field'])[index]
+            data.append({
+                'value': clean_punctuation(subfield, punct, spaced_punct),
+                'language': self.get_language_script(alt_gr['script'])
+            })
+        except Exception as err:
```
Comment (jma) [inline rero_ils/dojson/utils.py:290]: Really? Is it not too dangerous to ignore the errors?

File: rero_ils/dojson/utils.py [hunk 2]
```diff
@@ -532,6 +562,105 @@ def build_variant_title_data(self, string_set):
...
+    """Specialized Overdo for UNIMARC.
+
+    This class adds UNIMARC properties and functions to the ReroIlsOverdo.
+    """
+
+    bib_id = ''
+    lang_from_101 = None
+    alternate_graphic = {}
+
+    def __init__(self, bases=None, entry_point_group=None):
```
Comment (jma) [inline rero_ils/dojson/utils.py:575]: params and returns are missing in all methods of this class.

File: rero_ils/dojson/utils.py [hunk 3]
```diff
@@ -532,6 +562,105 @@ def build_variant_title_data(self, string_set):
...
+    def do(self, blob, ignore_missing=True, exception_handlers=None):
+        """Translate blob values and instantiate new model instance."""
+        self.count += 1
+        result = None
+        try:
+            self.blob_record = blob
+            try:
+                self.bib_id = self.get_fields(tag='001')[0]['data']
+            except Exception as err:
+                self.bib_id = '???'
```
Comment (jma) [inline rero_ils/dojson/utils.py:590]: Seems a bit weird.

File: rero_ils/dojson/utils.py [hunk 4]
```diff
@@ -532,6 +562,105 @@ def build_variant_title_data(self, string_set):
...
+            fields_101 = self.get_fields(tag='101')
+            if fields_101:
+                field_101_a = self.get_subfields(fields_101[0], 'a')
+                field_101_g = self.get_subfields(fields_101[0], 'g')
+                if field_101_a:
+                    self.lang_from_101 = field_101_a[0]
+                if field_101_g:
+                    self.lang_from_101 = field_101_g[0]
+
+            # self.init_lang()
```
Comment (jma) [inline rero_ils/dojson/utils.py:600]: to be removed?

File: rero_ils/dojson/utils.py [hunk 5]
```diff
@@ -532,6 +562,105 @@ def build_variant_title_data(self, string_set):
...
+
+        :param unimarc_script_code: the unimarc script code
+        :param unimarc_script_code: str
+        :return: language script code in the format `<lang_code>-<script_code>`
+        :rtype: str
+        """
+        if unimarc_script_code in UNIMARC_LANGUAGES_SCRIPTS:
+            script_code = UNIMARC_LANGUAGES_SCRIPTS[unimarc_script_code]
+            if script_code in LANGUAGES_SCRIPTS:
+                if self.lang_from_101 in LANGUAGES_SCRIPTS[script_code]:
```
Comment (jma) [inline rero_ils/dojson/utils.py:656]: multiple ifs can be replaced by:
```
script code = UNIMARC_LANGUAGES_SCRIPTS.get(unimarc_script_code)
if LANGUAGES_SCRIPTS.get(script_code, {}).get(self.lang_from_101):
```

File: rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py [hunk 6]
```diff
@@ -297,10 +327,10 @@ def build_place():
         # TODO: dates from 100 not working !!!!
         # if ind2 in (' ', '_', '1'):
         #     # startDate: 100, pos. 9-12 endDate: 100, pos. 13-16
-        #     field_100 = unimarctojson.get_fields(tag='100')
```
Comment (jma) [inline rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py:255]: Is this commented code still useful?

**Comment by jma** [comment]
Please update your commit message: change list is not provided.

File: rero_ils/dojson/utils.py (hunk 3 above)
Comment (reropag) [reply rero_ils/dojson/utils.py:590]: ok, for logging purpose

**Pull request merged**